src/components/AppText.tsx

```diff
@@ -22,13 +22,16 @@ export interface AppTextProps {
   bold?: boolean
   semiBold?: boolean
   medium?: boolean
-  color?: keyof DefaultTheme['font'] | keyof DefaultTheme['global']
+  color?: keyof DefaultTheme['font'] | keyof DefaultTheme['global'] | string
```

Comment:
Should we allow this to be a generic string? Wouldn't this fail bellow (line 35), if I provide a random string instead of a key from the theme object?

We could have 2 fields: `color` which accepts a generic string, and `themeColor`?

And if we want to be strict with regard to typings, we could use discriminating union like: 

```ts
type ColorProps = {
  color: keyof DefaultTheme['font'] | keyof DefaultTheme['global'];
  themeColor?: never;
};

type ThemeColorProps = {
  color?: never;
  themeColor: keyof DefaultTheme['font'] | keyof DefaultTheme['global'];
};

type NoColorProps = {
  color?: never;
  themeColor?: never;
};

type ComponentProps = ColorProps | ThemeColorProps | NoColorProps;
```

Reply:
maybe instead of `themeColor` we could have `theme`

Reply:
Sth like this:

```ts
import { themes, ThemeType } from '~/style/themes'

type FontColor = keyof DefaultTheme['font']
type GlobalColor = keyof DefaultTheme['global']

export type AppTextProps = {
  bold?: boolean
  semiBold?: boolean
  medium?: boolean
  color?: FontColor | GlobalColor
  colorTheme?: ThemeType
  size?: number
}

export default styled.Text<AppTextProps>`
  color: ${({ color, theme, colorTheme }) => {
    const th = colorTheme ? themes[colorTheme] : theme

    return color ? th.font[color as FontColor] || th.global[color as GlobalColor] : th.font.primary
  }};
`
```

Reply:
Yes, this would work. In terms of naming, maybe `colorTheme` could be replaced by `forcedTheme` or sth like this.

Reply:
Hm, I like `colorTheme`, it directly connects to the `color` prop since they have the same prefix. Do you have a strong opinion about it?

Reply:
No strong opinion :)